src/main/kotlin/org/rust/ide/console/RsConsoleFolding.kt

```diff
@@ -0,0 +1,60 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.console;
+
+import com.intellij.execution.ConsoleFolding
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.util.text.StringUtil.pluralize
+import com.intellij.openapiext.isUnitTestMode
+import org.rust.cargo.project.model.cargoProjects
+import org.rust.cargo.project.workspace.PackageOrigin
+import org.rust.cargo.runconfig.filters.*
+import org.rust.lang.core.resolve.resolveStringPath
+import java.io.File
+import kotlin.math.max
+
+class RsConsoleFolding : ConsoleFolding() {
+    override fun getPlaceholderText(project: Project, lines: List<String>): String? {
+        if (lines.size < 2) return null
+
+        // each call has two lines - function name and source code location
+        val count = lines.size / 2
+        val callText = pluralize("call", count)
+        return "<$count internal $callText>"
+    }
+
+    override fun shouldFoldLine(project: Project, line: String): Boolean {
+        val backtraceFilters = project.cargoProjects.allProjects.map {
+            RsBacktraceItemFilter(project, it.workspace)
+        }
+        if (backtraceFilters.any {
+                val functionName = it.parseBacktraceRecord(line)?.functionName ?: return@any false
+                val func = FilterUtils.normalizeFunctionPath(functionName)
+                val workspace = it.workspace ?: return@any false
+                val (_, pkg) = resolveStringPath(func, workspace, project) ?: return@any true
+
+                pkg.origin != PackageOrigin.WORKSPACE
+            }) {
+            return true
+        }
+
+        val locationFilters = project.cargoProjects.allProjects.mapNotNull {
+            val dir = it.workspaceRootDir ?: it.rootDir
+            RegexpFileLinkFilter(project, dir ?: return@mapNotNull null, RsBacktraceFilter.LINE_REGEX)
+        }
+        return locationFilters.any {
+            val fileName = it.matchLine(line)?.groups?.get(1)?.value ?: return@any false
+            var platformNeutralName = fileName.replace(File.separatorChar, '/')
+            if (isUnitTestMode) {
```

Comment:
Let's drop this condition and adjust test cases instead.
Our default test workspace has custom layout for symplicity.
Consider that you have the following Cargo.toml:
```toml
[lib]
name = "test-package"
path = "lib.rs"

[[bin]]
name = "test-package"
path = "main.rs"
```

Reply:
Ooh, now it makes sense. Thanks.